yb-voyager/src/tgtdb/oracle.go

```diff
@@ -0,0 +1,496 @@
+	if err != nil {
+		return 0, fmt.Errorf("write sqlldr control file %q: %w", sqlldrControlFilePath, err)
+	}
+
+	sqlldrLogFileName := fmt.Sprintf("%s-%s.log", tableName, batchName)
+	sqlldrLogFilePath := fmt.Sprintf("%s/sqlldr/%s", exportDir, sqlldrLogFileName)
+	sqlldrLogFile, err := os.Create(sqlldrLogFilePath)
+	if err != nil {
+		return 0, fmt.Errorf("create sqlldr log file %q: %w", sqlldrLogFilePath, err)
+	}
+	defer sqlldrLogFile.Close()
+
+	// fmt.Println("Running sqlldr for file: ", sqlldrControlFilePath)
+	connectionString := db.getConnectionUri(db.tconf)
+	user := getValue(connectionString, "user")
+	password := getValue(connectionString, "password")
+	connectString := getValue(connectionString, "connectString")
+
+	oracleConnectionString := fmt.Sprintf("%s/%s@\"%s\"", user, password, connectString)
+
+	sqlldrArgs := fmt.Sprintf("userid=%s control=%s log=%s DIRECT=TRUE NO_INDEX_ERRORS=TRUE", oracleConnectionString, sqlldrControlFilePath, sqlldrLogFilePath)
+	// fmt.Println("Args: ", sqlldrArgs)
+
+	cmd := exec.Command("sqlldr", sqlldrArgs)
+	var outbuf bytes.Buffer
+	var errbuf bytes.Buffer
+	cmd.Stdout = &outbuf
+	cmd.Stderr = &errbuf
+	// fmt.Printf("cmd: %v\n", cmd)
+	err = cmd.Run()
+	log.Infof("sqlldr output: %s", outbuf.String())
+	log.Errorf("sqlldr error: %s", errbuf.String())
+
+	// find ORA-00001: unique constraint * violated in log file
+	pattern := regexp.MustCompile(`ORA-00001: unique constraint \(.+?\) violated`)
+	scanner := bufio.NewScanner(sqlldrLogFile)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if pattern.MatchString(line) {
+			err = db.recordEntryInDB(tx, batch, rowsAffected)
+			if err != nil {
+				err = fmt.Errorf("record entry in DB for batch %q: %w", batch.GetFilePath(), err)
+			}
+			return 0, err
+		}
+	}
+
+	var err2 error
+	rowsAffected, err2 = getRowsAffected(sqlldrLogFilePath)
+	if err2 != nil {
+		err2 = fmt.Errorf("error parsing log file: %v", err)
+		return 0, err2
+	}
+	if err != nil {
+		// fmt.Printf("Error running sqlldr: %v\n", err)
+		return rowsAffected, fmt.Errorf("run sqlldr: %w", err)
+	}
+
+	err = db.recordEntryInDB(tx, batch, rowsAffected)
+	if err != nil {
+		err = fmt.Errorf("record entry in DB for batch %q: %w", batch.GetFilePath(), err)
+	}
+
+	return rowsAffected, err
+}
+
+func (db *TargetOracleDB) recordEntryInDB(tx *sql.Tx, batch Batch, rowsAffected int64) error {
+	cmd := batch.GetQueryToRecordEntryInDB(db.tconf.TargetDBType, rowsAffected)
+	_, err := tx.ExecContext(context.Background(), cmd)
+	if err != nil {
+		fmt.Printf("Error running query: %v\n", err)
+		return fmt.Errorf("insert into %s: %w", BATCH_METADATA_TABLE_NAME, err)
+	}
+	return nil
+}
+
+func getRowsAffected(logFilePath string) (int64, error) {
+	logFile, err := os.Open(logFilePath)
+	if err != nil {
+		return 0, err
+	}
+	defer logFile.Close()
+	re := regexp.MustCompile(`(\d+) Rows successfully loaded.`)
+	scanner := bufio.NewScanner(logFile)
```

Comment:
In the output of the `sqlldr`, it outputs a successfully loaded row count. see below - 
```

SQL*Loader: Release 21.0.0.0.0 - Production on Wed Jul 19 17:59:42 2023
Version 21.5.0.0.0

Copyright (c) 1982, 2022, Oracle and/or its affiliates.  All rights reserved.

Path used:      Direct

Load completed - logical record count 6500000. 

Table ACCOUNTS2:
  6500000 Rows successfully loaded.             <===================

Check the log file:
  /home/centos/sqlldr-test/ulcase1.log
for more information about the load.
```
So no need to read the log file for that, you can simply check this regex in the `outbuf` above.

Reply:
Made this change